src/components/UserSelector.tsx

```diff
@@ -1,6 +1,20 @@
-import React from 'react';
+import React, { useState } from 'react';
+import { useUsers } from '../context/UsersContext';
+import { User } from '../types/User';
+
+type Props = {
+  selectedUser: User | null;
+  selectUser: (user: User) => void,
+};
+
+export const UserSelector: React.FC<Props> = ({ selectedUser, selectUser }) => {
+  const [open, setOpen] = useState(false);
+  const users = useUsers();
+
+  const handleOpen = () => {
+    setOpen(!open);
```

Comment:
pls use [prevState](https://blog.greenroots.info/react-hook-usestate-lazy-initialization-previous-state#heading-how-to-get-the-previous-state-data-and-use-it) for such cases